GetStream/stream-chat-android
# [3176] Apply fixed image dimensions

# Please review #4011 before reviewing this PR

The core changes from #4011 have been cherry picked into this PR, so ideally we shouldn't be asking for an unnecessary core review but rather merge the other one first and then have no core diff with develop.

### 🎯 Goal

Previously our image previews inside the message list in Compose were static, meaning they would retain the same dimensions regardless of the aspect ratio of the image they were displaying.

We can achieve a nicer looking UI by resizing the images according to their aspect ratio.

### 🛠 Implementation details

- Deprecate `StreamDimens.attachmentsContentImageHeight` in favor of `StreamDimens.attachmentsContentImageMaxHeight`
- Calculate the ratio of the image attachments inside `ImageAttachmentContent` and apply them while respecting the max given height of an image attachment preview dictated by `StreamDimens.attachmentsContentImageMaxHeight`

### 🎨 UI Changes

_Add relevant screenshots_

| Before | After |
| --- | --- |
| ![Screenshot_1660128326](https://user-images.githubusercontent.com/37080097/183883321-cdb7bfb7-2296-4453-891b-41ef3399d0f6.png) | ![Screenshot_1660128262](https://user-images.githubusercontent.com/37080097/183883416-5d5c979a-5c20-4b22-b3ce-1152aedadbe6.png) |

### 🧪 Testing

1. Launch the Compose Sapmle app
2. Log in as Amit, Jc, Filip, Thierry or Tommaso
3. Enter the channel `Fixed image sizes test channel`
4. Scroll and observe all photos, report if any photo is exceeding dimensions or being rendered in a poor way
5. Feel free to post photos with various aspect ratios to see how they load
6. Read the changelog to make sure all the changes are understandable to the customer.

<details>

<summary>Patch that lowers the maximum height of an image attachment to 200.dp </summary>

```
Index: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt b/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt
--- a/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt	(revision 0eae187d4fbe12b96bb7abc547034f084b3db96a)
+++ b/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamDimens.kt	(date 1660128687709)
@@ -262,7 +262,7 @@
             quotedMessageAttachmentEndPadding = 0.dp,
             groupAvatarInitialsXOffset = 1.5.dp,
             groupAvatarInitialsYOffset = 2.5.dp,
-            attachmentsContentImageMaxHeight = 600.dp,
+            attachmentsContentImageMaxHeight = 200.dp,
         )
     }
 }
```

</details>

1. Apply the patch
2. Repeat the steps above and notice that the photos are now displayed with a different maximum height
3. Report if anything is rendered wrong


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

[!gif](https://media3.giphy.com/media/OPMdqmdAdnSBcEbiMa/giphy.gif?cid=ecf05e47efwsaxv9jby6megq3ssnlggnhguoz6wld8vvsemo&rid=giphy.gif&ct=g)
